src/components/PopupWithForm.js

```diff
@@ -28,9 +28,18 @@ export default class PopupWithForm extends Popup {
         return values;
     }
 
-    _submitForm = () => {
+    _submitForm = (event) => {
+        event.preventDefault(); // Prevent the form from submitting normally
+        this.setLoading(true); // Disable the submit button and show loading text
         const inputValues = this._getInputValues();
-        this._handleFormSubmit(inputValues);
+        this._handleFormSubmit(inputValues, () => {
+            this.setLoading(false); // Re-enable the submit button after the form is handled
+            // You can add a timeout here if you want to keep the popup open for a while
+            setTimeout(() => {
+                this.close(); // Close the popup after a delay
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


So, you can't use both actions in one callback.